PR:Add Brand entity and integrate with ApplicationDbContext

Ecomont opened the pull request:
This pull request introduces support for managing brands in the application by adding a new `Brand` entity and updating the database context to include it.

Ecomont committed:
Add Brand entity and integrate with ApplicationDbContext

alejandrogomezl commented on src/AppForSEII2526.API/Data/ApplicationDbContext.cs:
```diff
@@ -7,5 +7,6 @@ namespace AppForSEII2526.API.Data;
 public class ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : IdentityDbContext<ApplicationUser>(options) {
     public DbSet<DeliveryAssignment> DeliveryAssignments { get; set; }
     public DbSet<PurchaseOrder> PurchaseOrders { get; set; }
+    public DbSet<Brand> Brands { get; set; }
```
Please, resolve merge conflicts with development

Ecomont committed:
Add PurchaseProduct entity and integrate with DbContext

Introduced the `PurchaseProduct` entity to represent products
associated with purchase orders. Added a `DbSet<PurchaseProduct>`
property to `ApplicationDbContext` to enable database operations
for this entity. The `PurchaseProduct` class includes properties
for `Id`, `PurchaseOrderId`, `ProductId`, `Price`, and `Quantity`,
with appropriate data annotations for validation and precision.

Ecomont committed:
Extend payment and delivery management features

Ecomont committed:
updating dbcontext

alejandrogomezl reviewed (approved):
Well done sweety!!
